tests/test_eval.py

```diff
@@ -20,3 +25,28 @@ def test_eval(self):
         output, _ = eval_task.finetuning_results(model, ["Regression"])
         self.assertEqual(len(output), 564)
         self.assertTrue("WorldCerealCropland_LogisticRegression_f1" in output)
+
+    def test_spatial_inference(
+        self,
+    ):
+        model = Presto.load_pretrained()
+
+        test_data = pd.read_parquet(data_dir / "worldcereal_testdf.parquet")[:20]
+        labels = [99] * len(test_data)  # 99 = No cropland
+        labels[:10] = [11] * 10  # 11 = Annual cropland
+        test_data["LANDCOVER_LABEL"] = labels
+
+        spatial_data_prefix = "belgium_good_2020-12-01_2021-11-30"
+        spatial_data = rioxarray.open_rasterio(
+            data_dir / f"inference_areas/{spatial_data_prefix}.nc", decode_times=False
+        )
+        ground_truth_one_timestep = spatial_data.worldcereal_cropland.values[0, :, :]
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            eval_task = WorldCerealEval(test_data, test_data, Path(tmpdirname))
+            finetuned_model = eval_task._construct_finetuning_model(model)
+            eval_task.spatial_inference(finetuned_model, None)
+            output = xr.open_dataset(Path(tmpdirname) / f"{spatial_data_prefix}_finetuning.nc")
+            # np.flip because of how lat lons are stored vs x y
+            self.assertTrue(
```

Comment:
This tests if "ground truth" is loaded correctly, but we don't test if the spatial inference itself returns the expected results? Is that what we want?

Reply:
Testing the inference is a bit more complicated, since we don't know for sure what the expected values are. This lets us test the data loading (and turning the batches back into an `xr.Dataset`) is happening correctly. 

Ideally we would test the inference too, but visually it does seem to make sense and I am not certain how I would test that.

Reply:
Agreed, but I was just thinking if it makes sense to not only compare ground truth, but also the (flattened) predictions of spatial inference. Just to make sure the pipeline works as expected. We could do that just using the pretrained model, because the exact values of the predictions don't matter, as long as we verify that that they always return the same result?